codegen/src/main/java/software/amazon/awssdk/codegen/emitters/tasks/AuthSchemeGeneratorTasks.java

```diff
@@ -34,12 +36,16 @@ public AuthSchemeGeneratorTasks(GeneratorTaskParams dependencies) {
     }
 
     @Override
-    protected List<GeneratorTask> createTasks() throws Exception {
+    protected List<GeneratorTask> createTasks() {
+        AuthSchemeSpecUtils authSchemeSpecUtils = new AuthSchemeSpecUtils(model);
         List<GeneratorTask> tasks = new ArrayList<>();
         tasks.add(generateParamsInterface());
         tasks.add(generateProviderInterface());
         tasks.add(generateDefaultParamsImpl());
         tasks.add(generateDefaultProviderImpl());
+        if (authSchemeSpecUtils.useEndpointBasedAuthProvider()) {
+            tasks.add(generateEndpointBasedProvider());
+        }
```

Comment:
maybe the generate method name should also be changed to `generateModelBasedProvider`. Think through if any other codegen methods/classes should be renamed.